clairmeta/dcp_utils.py

```diff
@@ -106,6 +106,21 @@ def get_reel_for_asset(cpl, uuid):
         if uuid in uuids:
             return reel
 
+def get_first_reel_for_asset_type(cpl, type):
+    """ First reel of asset type lookup.
+
+        Args:
+            cpl (dict): Dictionary representation of CompositionPlayList.
+            type (str): Type of asset.
+
+        Returns:
+            Returns the first reel containing an asset of type
```

Comment:
Would it be more useful / generic to return the actual reel here and query the Position key in `check_subtitle_cpl_first_tt_event` ?

Reply:
We need to know if we are checking he first reel which contain the timed text. This is not neccecarily the first reel of the composition.

E.g. ST 429-2:2020 has this:

> ### 9.8 Timed Text Tracks
> A timed text track is established by the presence of a timed text asset (e.g. `MainSubtitle`, `MainCaption`, `ClosedSubtitle`, or `ClosedCaption`) in at least one Reel of a Composition. Once a timed text asset appears in one Reel, the established track shall be assumed to exist for the entire Composition, even if related timed text Asset elements are not present in all Reels.

This is also often seen for Interop. E.g. when a subtitle-less local distributor logo reel is added to a feature which has subtitles.

Reply:
Yes I agree with all that.

My comment was just that this function could be more useful (if the need for such lookup come up later on) if it returned the `dict` representing the Reel instead of just it's `Position` index. The behaviour would still be the same, exactly as you just described ;)